server/jetstream_api.go

```diff
@@ -1887,12 +1978,12 @@ func (s *Server) jsStreamInfoRequest(sub *subscription, c *client, a *Account, s
 			if js.isLeaderless() {
 				resp.Error = NewJSClusterNotAvailError()
 				// Delaying an error response gives the leader a chance to respond before us
-				s.sendDelayedAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp), nil)
+				s.sendDelayedAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp), nil, errRespDelay)
```

Comment:
@neilalexander Since this was based on your PR, I kept that, but do you have plans in the future to have a different deadline? It seems that we always pass the constant as of now.

If we don't have plans to have different values, then this (your) PR could be simplified. We could set the deadline after the pop() (in the go routine handler - granted, that would disregard the time from push to pop, but do we really care since the goal is to delay the responses?), and there would not be a need for a list/sort since they would naturally be ordered.

Reply:
The plan was to be able to use increasing deadlines for the rate limiting responses, yep. I just used `errRespDelay` in the current cases because that's the behaviour they have today.

Reply:
@neilalexander Ok. How do you want to proceed? Using this PR or yours? If yours, please use the test from this PR and you will likely need to fix some of the issues in yours (I can comment on there). If you want to use this one, let me know and I will remove the Draft status.

Reply:
I am planning to properly comb through this but overall I think your approach is good and certainly should be lighter on allocations/copies, so I'm very happy to close my PR and we can focus on this one.